typings/styled-components.d.ts

```diff
@@ -0,0 +1,162 @@
+import * as React from "react";
+import { StatelessComponent } from "react";
+
+type ConstrainedProps<C, P> = C & ({ defaultProps?: P } | { new(props?: P, context?: any): any });
+type StyledProps<P> = P & { theme: any };
+
+interface StyledFunction<T, P> {
+  (strs: TemplateStringsArray, ...fns: Array<(props: P) => string>): T;
```

Comment:
TypeScript doesn't seem to be allowing syntax for tagged string interpolation with generic type arguments ```styled.a<{ newProp: number; }>` ... ${ (props) => props.newProp } ... `; ```
Filed [an issue there](https://github.com/Microsoft/TypeScript/issues/13397)